### ISSUE TITLE
[bug]: InvokeAI fails to start generating from web interface

### ISSUE THREAD

**Issue by Sirachosur** [invoke-ai/InvokeAI#4440]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

Linux

### GPU

cuda

### VRAM

12Gb

### What version did you experience this issue on?

3.1.0

### What happened?

Used upgrade to change from 3.0.0 branch to 3.1.0, never worked ever since.

I upgraded the OS thinking it was a python issue (it wasn't)

I created a new user and reinstalled from the zip file to no avail.  

./invoke.sh ends with the following when I use (1) from the menu:
`Generate images with a browser-based interface
>> patchmatch.patch_match: ERROR - patchmatch failed to load or compile (libopencv_gapi.so.408: cannot open shared object file: No such file or directory).
>> patchmatch.patch_match: INFO - Refer to https://invoke-ai.github.io/InvokeAI/installation/060_INSTALL_PATCHMATCH/ for installation instructions.
[2023-09-02 09:43:08,898]::[InvokeAI]::INFO --> Patchmatch not loaded (nonfatal)
/home/invokeai/invokeai/.venv/lib64/python3.9/site-packages/torchvision/transforms/functional_tensor.py:5: UserWarning: The torchvision.transforms.functional_tensor module is deprecated in 0.15 and will be **removed in 0.17**. Please don't rely on it. You probably just need to use APIs in torchvision.transforms.functional or in torchvision.transforms.v2.functional.
  warnings.warn(
Traceback (most recent call last):
  File "/home/invokeai/invokeai/.venv/bin/invokeai-web", line 5, in <module>
    from invokeai.app.api_app import invoke_api
  File "/home/invokeai/invokeai/.venv/lib64/python3.9/site-packages/invokeai/app/api_app.py", line 26, in <module>
    from .api.dependencies import ApiDependencies
  File "/home/invokeai/invokeai/.venv/lib64/python3.9/site-packages/invokeai/app/api/dependencies.py", line 15, in <module>
    from invokeai.app.services.images import ImageService, ImageServiceDependencies
  File "/home/invokeai/invokeai/.venv/lib64/python3.9/site-packages/invokeai/app/services/images.py", line 37, in <module>
    from invokeai.app.util.metadata import get_metadata_graph_from_raw_session
  File "/home/invokeai/invokeai/.venv/lib64/python3.9/site-packages/invokeai/app/util/metadata.py", line 6, in <module>
    from invokeai.app.services.graph import Edge
  File "/home/invokeai/invokeai/.venv/lib64/python3.9/site-packages/invokeai/app/services/graph.py", line 154, in <module>
    class GraphInvocationOutput(BaseInvocationOutput):
  File "/home/invokeai/invokeai/.venv/lib64/python3.9/site-packages/invokeai/app/invocations/baseinvocation.py", line 618, in wrapper
    cls.__annotations__.update({"type": output_type_annotation})
AttributeError: type object 'GraphInvocationOutput' has no attribute '__annotations__'`

I don't exactly know how to start a trace to trap exactly where the process fails but the traceback seems clear enough.

Thanks for your time.

### Screenshots

![Backtrace](https://github.com/invoke-ai/InvokeAI/assets/77683206/d09257cb-8d9c-4927-aa06-64be5dce67f6)

### Additional context

uname -a returns 

`Linux stablediffusion 5.14.0-284.25.1.el9_2.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Aug 2 14:53:30 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux`

invoke.sh is called using the --host and the --port switches.

**Comment by czkoko** [comment]
The same error was encountered.
upgrade from 3.0 to 3.1
OS: MacOS 14 arm64
py: 3.9

**Comment by dannyvfilms** [comment]
Same issue on MacOS Sonoma 14.0 Public Beta (23A5328b). Issue is not present in separate install of 3.0.2post1.

```
Generate images with a browser-based interface
/Applications/InvokeAI_3.1.0/.venv/lib/python3.9/site-packages/torchvision/transforms/functional_tensor.py:5: UserWarning: The torchvision.transforms.functional_tensor module is deprecated in 0.15 and will be **removed in 0.17**. Please don't rely on it. You probably just need to use APIs in torchvision.transforms.functional or in torchvision.transforms.v2.functional.
  warnings.warn(
>> patchmatch.patch_match: INFO - Compiling and loading c extensions from "/Applications/InvokeAI_3.1.0/.venv/lib/python3.9/site-packages/patchmatch".
>> patchmatch.patch_match: ERROR - patchmatch failed to load or compile (Command 'make clean && make' returned non-zero exit status 2.).
>> patchmatch.patch_match: INFO - Refer to https://invoke-ai.github.io/InvokeAI/installation/060_INSTALL_PATCHMATCH/ for installation instructions.
[2023-09-02 11:10:00,599]::[InvokeAI]::INFO --> Patchmatch not loaded (nonfatal)
Traceback (most recent call last):
  File "/Applications/InvokeAI_3.1.0/.venv/bin/invokeai-web", line 5, in <module>
    from invokeai.app.api_app import invoke_api
  File "/Applications/InvokeAI_3.1.0/.venv/lib/python3.9/site-packages/invokeai/app/api_app.py", line 26, in <module>
    from .api.dependencies import ApiDependencies
  File "/Applications/InvokeAI_3.1.0/.venv/lib/python3.9/site-packages/invokeai/app/api/dependencies.py", line 15, in <module>
    from invokeai.app.services.images import ImageService, ImageServiceDependencies
  File "/Applications/InvokeAI_3.1.0/.venv/lib/python3.9/site-packages/invokeai/app/services/images.py", line 37, in <module>
    from invokeai.app.util.metadata import get_metadata_graph_from_raw_session
  File "/Applications/InvokeAI_3.1.0/.venv/lib/python3.9/site-packages/invokeai/app/util/metadata.py", line 6, in <module>
    from invokeai.app.services.graph import Edge
  File "/Applications/InvokeAI_3.1.0/.venv/lib/python3.9/site-packages/invokeai/app/services/graph.py", line 154, in <module>
    class GraphInvocationOutput(BaseInvocationOutput):
  File "/Applications/InvokeAI_3.1.0/.venv/lib/python3.9/site-packages/invokeai/app/invocations/baseinvocation.py", line 618, in wrapper
    cls.__annotations__.update({"type": output_type_annotation})
AttributeError: type object 'GraphInvocationOutput' has no attribute '__annotations__'
dannyvfilms@Daniels-MacBook-Pro-2 ~ %```

**Comment by Millu** [comment]
Wondering if this a python3.9 issue - if you're able to, could you update to python 3.10 and see if you get the same behavior?

**Comment by psychedelicious** [comment]
This is resolved in the next release, it's a python 3.9 issue

**Comment by Sirachosur** [comment]
Just installed python 3.11, set it as default and detroyed my .venv environnement.
Running the installer right now, I'll keep you posted if the workaround works.

**Comment by Sirachosur** [comment]
I confirm that upgrading python to 3.11 fixed the issue (after rebuilding the environment, that is).

This can be considered a workaround until the code is fixed.

**Comment by psychedelicious** [comment]
To be clear, the issue was resolved in #4413, which will be included in the next release.

**Comment by Sirachosur** [comment]
Seems that I'll have to peel my eyes properly next time, this is a duplicate.
